### PR TITLE
Wrap delta action in the response action when responseFormat=delta

### PR DESCRIPTION
**Pull request by linzhou-db**
Wrap delta action in the response action when responseFormat=delta

### CODE REVIEW

File: server/src/main/scala/io/delta/standalone/internal/model.scala [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.standalone.internal
+
+import io.delta.standalone.internal.actions.{
```
Comment (linzhou-db) [inline server/src/main/scala/io/delta/standalone/internal/model.scala:19]: This file contains the main change. 
The test changes are mostly boilerplate code changes.

File: server/src/main/scala/io/delta/standalone/internal/model.scala [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ * DeltaAddFile used in delta sharing protocol, copied from AddFile in delta.
+ *   Adding 4 delta sharing related fields: id/version/timestamp/expirationTimestamp.
+ *   If the client uses delta kernel, it should redact these fields as needed.
+ *       - id: used to uniquely identify a file, and in idToUrl mapping for executor to get
+ *             presigned url.
+ *       - version/timestamp: the version and timestamp of the commit, used to generate faked delta
+ *                            log file on the client side.
+ *       - expirationTimestamp: indicate when the presigned url is going to expire and need a
+ *                              refresh.
+ *   Ignoring 1 field: tags.
```
Comment (charlenelyu-db) [inline server/src/main/scala/io/delta/standalone/internal/model.scala:57]: Do we need to set `tags` to null when copy the file action?

**Review by charlenelyu-db — APPROVED** [review]

File: server/src/main/scala/io/delta/standalone/internal/model.scala (hunk 2 above)
Comment (linzhou-db) [reply server/src/main/scala/io/delta/standalone/internal/model.scala]: good catch, updated comment for all classes.

File: server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala [hunk 3]
```diff
@@ -261,17 +254,12 @@ class DeltaSharedTable(
       responseFormat: String,
       returnAddFileForCDF: Boolean = false): Object = {
     if (responseFormat == DeltaSharedTable.RESPONSE_FORMAT_DELTA) {
-      model.DeltaAddFile(
-        path = signedUrl.url,
+      DeltaResponseFileAction(
```
Comment (chakankardb) [inline server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala:257]: Update the comment headers

File: server/src/main/scala/io/delta/standalone/internal/model.scala [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+/**
+ * DeltaResponseFileAction used in delta sharing protocol. It wraps a delta action,
+ *   and adds 4 delta sharing related fields: id/version/timestamp/expirationTimestamp.
+ *       - id: used to uniquely identify a file, and in idToUrl mapping for executor to get
+ *             presigned url.
+ *       - version/timestamp: the version and timestamp of the commit, used to generate faked delta
+ *                            log file on the client side.
+ *       - expirationTimestamp: indicate when the presigned url is going to expire and need a
+ *                              refresh.
+ *   Suggest to redact the tags field before returning if there are sensitive info.
```
Comment (chakankardb) [inline server/src/main/scala/io/delta/standalone/internal/model.scala:49]: This header comment is cleaner than the DBR version. Consider copying it there.

File: server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala (hunk 3 above)
Comment (linzhou-db) [reply server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala:254]: done.

File: server/src/main/scala/io/delta/standalone/internal/model.scala (hunk 4 above)
Comment (linzhou-db) [reply server/src/main/scala/io/delta/standalone/internal/model.scala:50]: done.

**Review by chakankardb — APPROVED** [review]

**Pull request merged**